### PR TITLE
Fix [WebApi]Dynamic property name is null use convention routing

### PR DESCRIPTION
**Pull request by LianwMS**
Fix the issues

### CODE REVIEW

File: OData/src/System.Web.OData/OData/Routing/Conventions/DynamicPropertyRoutingConvention.cs [hunk 1]
```diff
@@ -99,6 +100,9 @@ public class DynamicPropertyRoutingConvention : NavigationSourceRoutingConventio
 
                 controllerContext.RouteData.Values[ODataRouteConstants.DynamicProperty] =
                     dynamicPropertSegment.PropertyName;
+                controllerContext.RouteData.Values[ODataParameterValue.ParameterValuePrefix + ODataRouteConstants.DynamicProperty] =
+                        new ODataParameterValue(dynamicPropertSegment.PropertyName,
+                            EdmLibHelpers.GetEdmPrimitiveTypeReferenceOrNull(typeof(string)));
```
Comment (xuzhg) [inline OData/src/System.Web.OData/OData/Routing/Conventions/DynamicPropertyRoutingConvention.cs]: always nullable-string?

Comment (xuzhg) [inline OData/src/System.Web.OData/OData/Routing/Conventions/DynamicPropertyRoutingConvention.cs]: Can you add one simple E2E test case  to verify the change works on [FromODataUri]?

**Comment by xuzhg** [comment]
I put two comments on the change. Please resolve it. Others looks good to me. Thanks.

**Comment by LianwMS** [comment]
Submit with 6cd30c8bccd730e410366cfaa5bcc1368a0a2ea5

**Pull request closed without merging**